src/useObserver.ts

```diff
@@ -24,41 +34,85 @@ export function useObserver<T>(
     const wantedForceUpdateHook = options.useForceUpdate || useForceUpdate
     const forceUpdate = wantedForceUpdateHook()
 
-    const reaction = useRef<Reaction | null>(null)
-    const committed = useRef(false)
+    // StrictMode/ConcurrentMode/Suspense may mean that our component is
+    // rendered and abandoned multiple times, so we need to track leaked
+    // Reactions.
+    const reactionTrackingRef = useRef<IReactionTracking | null>(null)
 
-    if (!reaction.current) {
-        // First render for this component. Not yet committed.
-        reaction.current = new Reaction(`observer(${baseComponentName})`, () => {
-            // Observable has changed. Only force an update if we've definitely
-            // been committed.
-            if (committed.current) {
-                forceUpdate()
-            }
-        })
+    if (!reactionTrackingRef.current) {
+        // First render for this component (or first time since a previous
+        // reaction from an abandoned render was disposed).
+        const trackingData: IReactionTracking = {
+            cleanAt: Date.now() + CLEANUP_LEAKED_REACTIONS_AFTER_MILLIS,
```

Comment:
Can we hide this into a factory function exported from the reaction tracking? Feels like exposing implementation detail here. It can simply accept the reaction instance and returning the `IReactionTracking`.

Reply:
I agree - importing the constant felt a bit icky when I wrote it.

Passing the reaction instance is a little tricky as we reference the IReactionTracking from the reaction callback. I'll see what I can do.

Reply:
Done. `const` + forward reference and all is well.